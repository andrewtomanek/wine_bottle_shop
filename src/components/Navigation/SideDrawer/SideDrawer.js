import React, { Fragment } from "react";
import { reduceCartState } from "../../../shared/helpers";
import Logo from "../../Logo/Logo";
import NavigationItems from "../NavigationItems/NavigationItems";
import classes from "./SideDrawer.module.css";
import Backdrop from "../../UI/Backdrop/Backdrop";

const sideDrawer = (props) => {
  let attachedClasses = [classes.SideDrawer, classes.Close];
  if (props.open) {
    attachedClasses = [classes.SideDrawer, classes.Open];
  }

  const calculateCounter = (inventory) => {
    if (reduceCartState(inventory) === 0) return null;
    return (
      <span
        onMouseEnter={() => props.showCartDrawer(true)}
        className={classes.CartCounter}
      >
        {reduceCartState(inventory)}
      </span>
    );
  };

  return (
    <Fragment>
      <Backdrop show={props.open} clicked={props.closed} />
      <div className={attachedClasses.join(" ")} onClick={props.closed}>
        <div className={classes.Logo}>
          <Logo />
        </div>
        <nav className={classes.Navigation}>
          <NavigationItems isAuthenticated={props.isAuth} />
        </nav>
      </div>
    </Fragment>
  );
};

export default sideDrawer;
