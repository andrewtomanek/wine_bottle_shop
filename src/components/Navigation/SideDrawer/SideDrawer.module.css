.SideDrawer {
  display: none;
}

.CartCounter {
  color: white;
  padding: 0.5rem 1.5rem;
  font-weight: 700;
  font-size: 2rem;
  border-bottom: 0.2rem solid transparent;
  background-color: var(--red);
  text-decoration: none;
}

.Navigation {
  padding: 2rem 0;
}

@media (max-width: 600px) {
  .SideDrawer {
    grid-area: 1/1/4/1;
    display: grid;
    grid-template-rows: 10vh 1fr;
    grid-template-columns: 100%;
    align-items: center;
    justify-content: center;
    grid-gap: 0.5rem;
    margin: 0;
    height: 100vh;
    z-index: 200;
    background-color: white;
    padding: 1rem;
    box-sizing: border-box;
    transition: transform 0.3s ease-out;
  }
  .Open {
    transform: translateX(0);
  }

  .Close {
    transform: translateX(-100vw);
  }

  .Logo {
    height: 100%;
  }
}
