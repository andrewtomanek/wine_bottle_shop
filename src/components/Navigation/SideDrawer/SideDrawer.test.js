import React from "react";
import { shallow } from "enzyme";
import SideDrawer from "./SideDrawer";

const closed = jest.fn();

const props = {
  isAuth: true,
  open: true,
  closed,
};

describe("Renders SideDrawer", () => {
  it("renders logo", () => {
    const wrapper = shallow(<SideDrawer {...props} />);
    expect(wrapper.find(".Logo").text()).toEqual("<logo />");
  });

  it("renders NavigationItems", () => {
    const wrapper = shallow(<SideDrawer {...props} />);
    expect(wrapper.find("nav").text()).toEqual("<NavigationItems />");
  });

  it("renders SideDrawer component", () => {
    const wrapper = shallow(<SideDrawer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
