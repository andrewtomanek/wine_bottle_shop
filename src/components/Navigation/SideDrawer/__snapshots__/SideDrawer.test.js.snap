// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders SideDrawer renders SideDrawer component 1`] = `
<Fragment>
  <backdrop
    clicked={[MockFunction]}
    show={true}
  />
  <div
    className="SideDrawer Open"
    onClick={[MockFunction]}
  >
    <div
      className="Logo"
    >
      <logo />
    </div>
    <nav
      className="Navigation"
    >
      <NavigationItems
        isAuthenticated={true}
      />
    </nav>
  </div>
</Fragment>
`;
